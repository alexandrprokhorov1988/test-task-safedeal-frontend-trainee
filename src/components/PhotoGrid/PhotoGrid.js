import React from 'react';
import './PhotoGrid.css';
import PropTypes from "prop-types";
import Image from '../../components/Image/Image';

function PhotoGrid({ cards, onImageClick }) {

  return (
    <section className="photo-grid">
      {cards.map((card) => (
        <Image
          key={card.id}
          id={card.id}
          url={card.url}
          onImageClick={onImageClick}
        />
      ))}
    </section>
  );
}

PhotoGrid.propTypes = {
  cards: PropTypes.arrayOf(PropTypes.object).isRequired,
  onImageClick: PropTypes.func.isRequired,
};

export default PhotoGrid;
