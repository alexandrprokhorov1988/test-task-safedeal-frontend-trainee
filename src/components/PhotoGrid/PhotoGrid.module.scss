.photoGrid {
  margin: 0 auto;
  max-width: 728px;
  width: 95%;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(229px, 1fr));
  grid-row-gap: 30px;
  grid-column-gap: 20px;
  padding-top: 30px;
  padding-bottom: 377px;
  box-sizing: border-box;
}
