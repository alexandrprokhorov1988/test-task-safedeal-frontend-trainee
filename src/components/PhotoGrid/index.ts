export {default} from './PhotoGrid';
export {default as PhotoGridImage} from './PhotoGridImage';
