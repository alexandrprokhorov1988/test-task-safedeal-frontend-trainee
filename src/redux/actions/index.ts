import {setIsLoading, setOriginSizeImage, setCards, setComments} from './photoGridActions';
import {setIsLoadingComment} from './popupActions';
import {setIsOpenPopup} from './mainActions';

export default {
  setIsLoading,
  setOriginSizeImage,
  setCards,
  setComments,
  setIsLoadingComment,
  setIsOpenPopup
}
