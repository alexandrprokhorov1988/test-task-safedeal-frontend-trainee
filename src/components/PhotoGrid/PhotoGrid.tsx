import React from 'react';
import {observer} from "mobx-react-lite";
import './PhotoGrid.css';
import PhotoGridImage from './PhotoGridImage';
import Preloader from '../Preloader';
import photoGrid from '../../stores/photoGridStore/photoGrid';
import popup from '../../stores/popupStore/popup';

type PhotoGridProps = {
  /**
   * PhotoGrid handleOpen.
   */
  onOpen: () => void;
}

/**
 * PhotoGrid component.
 */
const PhotoGrid: React.FC<PhotoGridProps> = observer(({onOpen}) => {

  console.log('photogrid');// todo del

  function handleGetOriginalSizeImage(imageId: number) {
    if (imageId === popup.originSizeImage.id) {
      onOpen();
    } else {
     popup.getOriginSizeImage(imageId, onOpen);
    }
    return true;
  }

  React.useEffect(() => {
    console.log('запрос на сервер за карточками'); // todo del
    photoGrid.getInitialImages();
  }, []);

  return (
    <section className="photo-grid">
      {photoGrid.isLoading ? <Preloader/> :
        <>
          {photoGrid.cards && photoGrid.cards.map((card: any) => (
            <PhotoGridImage
              key={card.id}
              id={card.id}
              url={card.url}
              onImageClick={handleGetOriginalSizeImage}
            />
          ))}
        </>
      }
    </section>
  );
});

export default React.memo(PhotoGrid);
