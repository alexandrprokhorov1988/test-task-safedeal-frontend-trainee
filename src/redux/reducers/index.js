import {combineReducers} from 'redux';
import {mainReducer} from './main';
import {photoGridReducer} from './photoGrid';
import {popupReducer} from './popup';

export const rootReducer = combineReducers({
  main: mainReducer,
  photoGrid: photoGridReducer,
  popup: popupReducer,
});
