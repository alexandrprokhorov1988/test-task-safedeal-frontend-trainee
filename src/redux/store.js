import {applyMiddleware, createStore} from 'redux'
import {rootReducer} from './reducers';
import {logger} from "./middlewares/logger";

export const store = createStore(rootReducer, applyMiddleware(logger));
