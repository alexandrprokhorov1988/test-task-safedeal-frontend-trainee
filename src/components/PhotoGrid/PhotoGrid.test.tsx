import React from 'react';
import {render, unmountComponentAtNode} from "react-dom";
import renderer from 'react-test-renderer';
import {act} from "react-dom/test-utils";
import PhotoGrid from './PhotoGrid';
import {combineReducers, createStore} from "redux";
import {Provider} from "react-redux";

let container: any = null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('PhotoGrid component', () => {
  it('PhotoGrid snapshot', () => {
    const store = createStore(combineReducers({
      main: () => ({
        isOpenPopup: false,
      }),
      photoGrid: () => ({
        cards: [{
          id: 1,
          url: "#/testUrl_1",
        },
          {
            id: 2,
            url: "#/testUrl_2",
          }],
        isLoading: false,
        currentOriginSizeImage: {id: 5, url: '#/testUrl'},
        comments: [{id: 1, text: 'Тестовый коммент', date: '55.55.2055'}],
      }),
      popup: () => ({
        isLoadingComment: false,
      }),
    }));

    const result = renderer
      .create(<Provider store={store}><PhotoGrid onOpen={() => {
      }}/></Provider>)
      .toJSON();
    expect(result).toMatchSnapshot();
  });
});
