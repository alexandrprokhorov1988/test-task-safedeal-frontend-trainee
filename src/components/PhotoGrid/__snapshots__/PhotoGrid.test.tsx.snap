// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoGrid component PhotoGrid snapshot 1`] = `
<section
  className="photo-grid"
>
  <img
    alt="Картинка"
    className="photo-grid-image"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    src="#/test1"
    tabIndex={0}
  />
  <img
    alt="Картинка"
    className="photo-grid-image"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="presentation"
    src="#/test5"
    tabIndex={0}
  />
</section>
`;
